src/libcore/convert.rs

```diff
@@ -30,6 +50,10 @@ use marker::Sized;
 ///
 /// [book]: ../../book/borrow-and-asref.html
 ///
+/// **Note:** these traits are for trivial conversion. **They must not fail**. If
```

Comment:
Do you mean that a panic could be acceptable in this case or do you mean that implementing `Into<Option<T>>` is not idiomatic ?